src/parser/expression/columnref_expression.cpp

```diff
@@ -51,6 +51,19 @@ const string &ColumnRefExpression::GetTableName() const {
 	return column_names[0];
 }
 
+const string &ColumnRefExpression::GetSchemaName() const {
+	D_ASSERT(column_names.size() >= 3 && column_names.size() <= 4);
+	if (column_names.size() == 4) {
+		return column_names[1];
```

Comment:
We have a very flexible dot syntax - we support all of `schema.table.column`, `catalog.table.column`, `table.column.struct_entry`, `column.struct_entry.nested_struct_entry`. As such `GetSchemaName()` and `GetCatalogName()` as they are implemented here are rather confusing and can be wrong.

Could you remove these from the `ColumnRefExpression`?